packages/valory/skills/mech_interact_abci/models.py

```diff
@@ -161,22 +184,17 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self.escrow_payment_condition_address = self._ensure(
             "escrow_payment_condition_address", kwargs, str
         )
-        self.token_address = self._ensure("token_address", kwargs, str)
-        self.plan_fee_nvm = self._ensure("plan_fee_nvm", kwargs, int)
-        self.plan_price_mech = self._ensure("plan_price_mech", kwargs, int)
-        self.subscription_credits = self._ensure("subscription_credits", kwargs, int)
-        self.subscription_nft_address = self._ensure(
-            "subscription_nft_address", kwargs, str
-        )
-        self.nft_sales_address = self._ensure("nft_sales_address", kwargs, str)
-        # only required for base chain (usdc is the payment token for subscription)
-        self.subscription_token_address = self._ensure(
-            "subscription_token_address", kwargs, str
-        )
-        self.subscription_provider_address = self._ensure(
-            "subscription_provider_address", kwargs, str
-        )
-        self.did = self._ensure("did", kwargs, str)
+
+        nvm_config = CHAIN_TO_NVM_CONFIG[str(self.mech_chain_id)]
+        self.plan_fee_nvm = nvm_config["plan_fee_nvm"]
+        self.plan_price_mech = nvm_config["plan_price_mech"]
+        self.subscription_credits = nvm_config["subscription_credits"]
+        self.subscription_nft_address = nvm_config["subscription_nft_address"]
+        self.nft_sales_address = nvm_config["nft_sales_address"]
+        self.subscription_token_address = nvm_config["subscription_token_address"]
+        self.subscription_provider_address = nvm_config["subscription_provider_address"]
+        self.plan_did = nvm_config["did"]
+        self.did = self.plan_did.replace("did:nv:", "0x")
```

Comment:
Instead of blindly replacing, I would suggest supporting both inputs. If not starting with `0x`, then try to replace `did:nv`. If this is not available either, then fail with an incorrect format error. Moreover, I would suggest additional validation since we know the expected number of characters that follow. It's better to fail early than during runtime.